src/api/c/libspeechd.c

```diff
@@ -2020,17 +2008,44 @@ static char *get_param_str(char *reply, int num, int *err)
 
 	pos_begin = pos + 4;
 	pos_end = strstr(pos_begin, "\r\n");
-	if (pos_end == NULL) {
+	if (*pos_end == 0) {
```

Comment:
Noticing this just now: there shouldn't be a `*`, it's really the pointer that needs to be tested. You can leave the code as it is.